src/ThriveStakingNative.sol

```diff
@@ -0,0 +1,40 @@
+// SPDX-License-Identifier: MIT
+pragma solidity ^0.8.24;
+
+import {ThriveStakingBase} from "./ThriveStakingBase.sol";
+
+contract ThriveStakingNative is ThriveStakingBase {
+    /// @notice Initialize for native token staking. The token address is set to address(0).
+    function initialize(
+        uint256 _yieldRate,
+        uint256 _minStakingAmount,
+        address _accessControlEnumerable,
+        bytes32 _role
+    ) public initializer {
+        _initialize(
+            address(0),
+            _yieldRate,
+            _minStakingAmount,
+            _accessControlEnumerable,
+            _role
+        );
+    }
+
+    /// @notice Stake native tokens by sending ETH along with the call.
+    function stake() external payable nonReentrant {
```

Comment:
I don't know if this overrides parent method as it's not virtual, IMO I would override just this:
```
    function _stake(uint256 amount) internal virtual override {
        require(
            amount == msg.value, "..."
        );
```
and make stake payable in base class.
Also in ThriveStakingIERC20 I would make _stake to have this condition
```
require(msg.value == 0, '...')
```